WordPress/Classes/ViewRelated/Me/My Profile/Change Username/ChangeUsernameViewController.swift

```diff
@@ -189,7 +148,50 @@ private extension ChangeUsernameViewController {
     }
 }
 
-fileprivate extension UIAlertController {
+private extension UIAlertController {
+    static var cancellableKey = 0
+
+    static func confirmationPrompt(forSelectedUsername selectedUsername: String, confirmed: @escaping () -> Void) -> UIAlertController {
+        typealias Constants = ChangeUsernameViewController.Constants
+        let alertController = UIAlertController(title: Constants.Alert.title, message: "", preferredStyle: .alert)
+        alertController.addAttributeMessage(String(format: Constants.Alert.message, selectedUsername),
+                                            highlighted: selectedUsername)
+        alertController.addCancelActionWithTitle(Constants.Alert.cancel, handler: { _ in
+            DDLogInfo("User cancelled alert")
+        })
+
+        let action = alertController.addDefaultActionWithTitle(Constants.Alert.change, handler: { [weak alertController] _ in
+            guard let alertController else { return }
+            guard let textField = alertController.textFields?.first,
+                textField.text == selectedUsername else {
+                    DDLogInfo("Username confirmation failed")
+                    return
+            }
+            DDLogInfo("User changes username")
+            confirmed()
+        })
+        action.isEnabled = false
+
+        alertController.addTextField { [weak action, unowned alertController] textField in
+            textField.placeholder = Constants.Alert.confirm
+
+            let cancellable = NotificationCenter.default
+                .publisher(for: UITextField.textDidChangeNotification, object: textField)
+                .sink(receiveValue: { [weak action, weak textField] _ in
+                    let enabled = textField?.text?.isEmpty == false && textField?.text == selectedUsername
+                    action?.isEnabled = enabled
+                    textField?.textColor = enabled ? .success : .text
+                })
+            // Bind the subscription with the lifetime of the alert. When the alert instance is released, `cancellable` will be
+            // released too which will automatically unregister the notification observer.
+            objc_setAssociatedObject(alertController, &UIAlertController.cancellableKey, cancellable, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
```

Comment:
@crazytonyli @jkmassel I generally prefer having an `extension` on `UIAlertController` that encapsulates all the configuration logic for a customize instance.

However, in this particular instance, the approach comes with the need for this legit but unusual `objc_setAssociatedObject` call.

All in all, I think the original version from #21003 might be best because of it's straightforwardness.

I'll leave it up to @jkmassel to decide. Also, I think given this address an relatively unfrequent use case (changing the user name) we don't need to rush into merging either version. Or, if we really want to rest assured that the bug is fixed ASAP, we could merge either for 22.8, and then refine if necessary.